OasisDEX/oasis-borrow
# 🧿 Ajna header selector

# [🧿 Ajna header selector](https://app.shortcut.com/oazo-apps/story/7327/select-collateral-token-drop-down)

~~For now, just the selector part without table and connection to anything.~~
Now also includes table with fake data randomized, but based on curated tokens list. For now missing parts are connection to the blockchain ~~and marking some token pairs as not yet available~~.

![image](https://user-images.githubusercontent.com/16230404/216308382-6d7da45d-7e5b-4ce2-a1ce-d1ddb65b92d5.png)

Additionally, `HeaderSelector` can be used in any context as it inherits font styles from its parent.

![image](https://user-images.githubusercontent.com/16230404/216021242-dbd072ef-1657-426a-acbb-58a0a76e2869.png)
  
## Changes 👷‍♀️

- Created `HeaderSelector` component,
- created `AjnaSelectorController` which could be reused for earn and multiply,
- added support for JSX to discover table - that should be done when I was working on user's position, but I was fixating myself on keeping that table pure data. Not anymore.
  
## How to test 🧪

- Go to `/ajna/open/borrow` and see how selector behaves. ~~Changing its value won't affect anything on that page (for now),~~
- ignore 4 Security Hotspots from sonarcloud - `Math.random` in `useEffect` is just temporary for dummy data.
